generated/nidaqmx/nidaqmx_service_registrar.cpp

```diff
@@ -0,0 +1,53 @@
+
+//---------------------------------------------------------------------
+// This file is automatically generated. All manual edits will be lost.
+//---------------------------------------------------------------------
+// Service registrar implementation for the NI-DAQMX Metadata
+//---------------------------------------------------------------------
+#include "nidaqmx_library.h"
+
+#include <grpcpp/server_builder.h>
+
+#include "nidaqmx_service.h"
+#include "nidaqmx_service_registrar.h"
+
+namespace nidaqmx_grpc {
+
+namespace {
+struct LibraryAndService {
+  LibraryAndService(
+    const std::shared_ptr<nidevice_grpc::SessionResourceRepository<TaskHandle>>& resource_repository,
+    const NiDAQmxFeatureToggles& feature_toggles) 
+      : library(), 
+      service(
+        &library, 
+        resource_repository, 
+        feature_toggles) {
+  }
+  NiDAQmxLibrary library;
+  NiDAQmxService service;
+};
+}
+
+std::shared_ptr<void> register_service(
```

Comment:
Note: `std::shared_ptr<void>` is being used like `std:any` here, but we can't use `any` because it's not supported in the gcc used for nilrt. Basically it's type-erasing the `LibraryAndService` class so that we can transfer lifetime ownership without an intrusive empty interface (and we don't actually want to expose a different interface to interact with the service once it's created).